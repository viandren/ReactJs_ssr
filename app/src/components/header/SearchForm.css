.search-form {
    top: 50%;
    width: 80%;
    margin: 0 auto;
}
.search-input {
    font-family: 'Roboto', sans-serif;
    color: #555555;
    font-size: 1.2rem;
    padding: 1rem 1.5rem;
    border-radius: 0.2rem;
    background-color: #424242;
    border: none;
    width: 60%;
    transition: all 0.3s;
    float: left;
    margin-right: 1rem;
}

.search-input:focus {
    color: #c7c1c1;
    border-color: #555555;
}

.search-button {
    font-size: 1.2rem;
    width: 15%;
    border: none;
    background-color: #F65261;
    color: #FFFFFF;
    padding: 1rem 1.5rem;
    border-radius: 0.2rem;
}

.search-result {
    display: none;
}
