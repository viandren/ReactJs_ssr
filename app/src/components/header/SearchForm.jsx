import './SearchForm.css';

import React from "react";
import { useState } from "react";


export default function SearchForm(props) {

  const [query, setQuery] = useState(props.defaultValue || '');

  const handleChange = e => {
    setQuery(e.target.value);
  };

  const handleSubmit = e => {
    e.preventDefault();
    props.handleSubmit(query);
  };

return (
    <div className="search-form" data-testid="searchForm">
      <form data-testid="form" onSubmit={handleSubmit}>
        <input placeholder={props.placeholderText} onChange={handleChange} name="searchInput" className="search-input" value={query}/>
        <button onClick={handleSubmit} type="submit" className="search-button">
          Search
        </button>
      </form>
    </div>
  )
}

