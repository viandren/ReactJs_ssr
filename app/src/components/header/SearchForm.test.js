import SearchForm from "./SearchForm.js";
import { render, screen, cleanup, fireEvent } from "@testing-library/react";



describe ("SearchForm", () => {

    afterEach(() => {
        cleanup(); 
    })

    test('SearchForm is rendered', () => {
        renderComponent((query) => {console.log("search initiated with query: " + query)});
        const searchForm = screen.getByTestId("searchForm");
        expect(searchForm).toBeInTheDocument();
    })

    test('Snapshot', () => {
        const {asFragment} = renderComponent((query) => {console.log("search initiated with query: " + query)});
        expect(asFragment()).toMatchSnapshot();
    })

    test('SearchForm and serach button are rendered', () => {
        renderComponent((query) => {console.log("search initiated with query: " + query)});
        
        const input = screen.getByRole('textbox')
        expect(input).toBeInTheDocument();
        const searchButton = screen.getByText('Search')
        expect(searchButton).toBeInTheDocument();
    })

    test('SearchForm typing and submit', () => {
        const onSearch = jest.fn();
        renderComponent(onSearch);
        
        const input = screen.getByRole('textbox')
        expect(input).toBeInTheDocument();
        fireEvent.change(input, { target: { value: 'whatever' } });
        expect(input).toHaveDisplayValue("whatever");
        const form = screen.getByTestId("form")
        fireEvent.submit(form);
    })

    test('SearchForm button clicked', () => {
        const onSearch = jest.fn();
        renderComponent(onSearch);
        
        const input = screen.getByRole('textbox')
        expect(input).toBeInTheDocument();
        const searchButton = screen.getByRole('button', 'Search')
        expect(searchButton).toBeInTheDocument();
        fireEvent.change(input, { target: { value: 'whatever and something' } });
        expect(input.value).toBe("whatever and something");
        fireEvent.click(searchButton);
    })
});

function renderComponent(onSearch) {
    return render(<SearchForm 
        placeholderText="horror"
        handleSubmit={onSearch}
        />);
}