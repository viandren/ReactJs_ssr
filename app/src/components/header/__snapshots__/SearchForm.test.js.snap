// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm Snapshot 1`] = `
<DocumentFragment>
  <div
    class="search-form"
    data-testid="searchForm"
  >
    <form
      data-testid="form"
    >
      <input
        class="search-input"
        name="searchInput"
        placeholder="horror"
        value=""
      />
      <button
        class="search-button"
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
</DocumentFragment>
`;
